src/contracts/core/AVSDirectory.sol

```diff
@@ -77,7 +184,11 @@ contract AVSDirectory is
         );
         require(
             delegation.isOperator(operator),
-            "AVSDirectory.registerOperatorToAVS: operator not registered to EigenLayer yet");
+            "AVSDirectory.registerOperatorToAVS: operator not registered to EigenLayer yet"
+        );
+        require(
+            !isOperatorSetAVS[msg.sender]
```

Comment:
`operatorAVSOperatorSetCount != 0`?

Reply:
Are you suggesting to only check `operatorAVSOperatorSetCount != 0`? The end result of doing so is that the AVS is not forced into becoming an operator set AVS, which we want to do here